### PR TITLE
Adding streaming_conv reference code.

### PR DESCRIPTION
**Pull request by cad-audio**
BUG=b/327502734

### CODE REVIEW

File: tensorflow/lite/micro/kernels/xtensa/conv_hifi.cc [hunk 1]
```diff
@@ -115,6 +115,73 @@ TfLiteStatus ConvPrepareHifi(TfLiteContext* context, TfLiteNode* node) {
...
+  /* TODO(b/277112516): Dilation is currently not supported on HiFi 4 NN Library
+   */
+  bool inputs_and_bias_ok = bias != nullptr;
+#if defined(HIFI3) || defined(HIFI4)
+  inputs_and_bias_ok =
+      inputs_and_bias_ok &&
+      (input->type == kTfLiteInt8 ||
+       (input->type == kTfLiteInt16 && bias->type == kTfLiteInt64));
+#else
+  inputs_and_bias_ok = inputs_and_bias_ok && (input->type == kTfLiteInt8);
```
Comment (rascani) [inline tensorflow/lite/micro/kernels/xtensa/conv_hifi.cc:149]: Or is this supposed to be for HIFI5?

File: tensorflow/lite/micro/kernels/xtensa/streaming_conv.cc [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  TfLiteEvalTensor* output =
+      tflite::micro::GetEvalOutput(context, node, kConvOutputTensor);
+  const TfLiteEvalTensor* filter =
+      tflite::micro::GetEvalInput(context, node, kConvWeightsTensor);
+  const TfLiteEvalTensor* bias =
+      tflite::micro::GetEvalInput(context, node, kConvBiasTensor);
+
+  switch (input->type) {
+    case kTfLiteInt16: {
+#if defined(HIFI3) || defined(HIFI4)
```
Comment (rascani) [inline tensorflow/lite/micro/kernels/xtensa/streaming_conv.cc:55]: HIFI5?

File: tensorflow/lite/micro/kernels/xtensa/streaming_conv.cc [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+      // Note that int32 bias is not widely supported and might be risky (e.g.
+      // http://b/262003750). As such, while we have a fallback to the reference
+      // implementation, production use-cases should only have int64 bias.
+      if (bias->type == kTfLiteInt32) {
+        return ConvReferenceEvalInt16(context, node);
```
Comment (rascani) [inline tensorflow/lite/micro/kernels/xtensa/streaming_conv.cc:60]: Should this be StreamingConvReferenceEvalInt16?

File: tensorflow/lite/micro/kernels/xtensa/streaming_conv.cc [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+      } else {
+        return StreamingConvEvalHifiInt16(context, node, params, op_data, input, filter,
+                                 bias, output);
+      }
+#else
+      return ConvReferenceEvalInt16(context, node);
```
Comment (rascani) [inline tensorflow/lite/micro/kernels/xtensa/streaming_conv.cc:66]: Should this be StreamingConvReferenceEvalInt16?

**Review by rascani — COMMENTED** [review]
Generally LGTM, just a few minor comments.

**Comment by cad-audio** [comment]
@rascani, added HiFi 5 changes as well.

**Review by rascani — APPROVED** [review]

**Comment by rascani** [comment]
@vnileshdaiict can you sign the CLA?

**Comment by vnileshdaiict** [comment]
Hi,
We have pushed the same changes with cad-audio account.
Please ignore the PR raised from this account.

Thanks and Regards,
Nileshkumar

**Comment by rascani** [comment]
This is the PR from cad-audio, but the CLA is required to be signed by all accounts that have a commit on the PR. Since you have a commit on the PR, the change cannot be merged until you sign the CLA.

**Comment by rascani** [comment]
@Mergifyio refresh

**Pull request merged**